kv/urm_test.go

```diff
@@ -49,3 +63,35 @@ func initUserResourceMappingService(s kv.Store, f influxdbtesting.UserResourceFi
 		}
 	}
 }
+
+func BenchmarkReadURMs(b *testing.B) {
+	urms := influxdbtesting.UserResourceFields{
+		UserResourceMappings: make([]*influxdb.UserResourceMapping, b.N),
+	}
+	idgen := snowflake.NewDefaultIDGenerator()
+	userCount := b.N / 50
+	if userCount < 1 {
+		userCount = 1
+	}
+	users := make([]influxdb.ID, userCount)
+	for i := 0; i < userCount; i++ {
+		users[i] = idgen.ID()
+	}
+
+	for i := 0; i < b.N; i++ {
```

Comment:
Just checked the docs and this loop should be around the actual function being tested.
namely: `svc.FindUserResourceMappings(...)`.
I tried adapting this benchmark to mine for both a warm and cold cache and it gave bizarre results.
the `b.N` value should be used to loop over and call the function being tested. Rather than populating the state it iterates over.
I changed my benchmark to populated with a large fixed amount of data and then looped over as described above and it reflected the reality I expected.